tests/acceptance/test_organization_integration.py

```diff
@@ -0,0 +1,150 @@
+        assert button.get_label() == label
+        if icon:
+            assert button.get_icon_href() == icon
+
+
+class OrganizationIntegrationSettingsTest(OrganizationIntegrationAcceptanceTestCase):
+    """
+    As a user(type?), I can setup, configure, and remove an integration.
+    """
+    # TODO(lb): Tests to be written
+    # test_setup_new_integration_with_repository
+    # test_setup_new_integration_with_issue_sync
+    # test_remove_existing_integration_installation
+    # test_update_legacy_integration
+    # test_user_permissions_for_integration_settings
+    # test_add_multiple_integrations_to_one_provider
+
+    def setUp(self):
+        super(OrganizationIntegrationSettingsTest, self).setUp()
+        self.org_integration_settings_path = u'/settings/{}/integrations/'.format(
+            self.organization.slug)
+
+        self.provider = mock.Mock()
+        self.provider.key = 'example'
+        self.provider.name = 'Example Installation'
+
+    def assert_can_create_new_installation(self):
+        self.load_page(self.org_integration_settings_path)
+        org_settings_page = OrganizationIntegrationSettingsPage(
+            browser=self.browser
+        )
+        provider_element = org_settings_page.get_provider(self.provider)
+
+        integration_details_modal = org_settings_page.click_install_button(provider_element)
+        integration_details_modal.add_button.click()
+        org_settings_page.click_through_integration_setup(
+            integration_details_modal,
+            ExampleIntegrationSetupWindowElement,
+            {'name': self.provider.name}
+        )
+
+        installation_element = provider_element.get_installation_with_name(self.provider.name)
+        assert installation_element
+        assert Integration.objects.filter(
+            provider=self.provider.key,
+            external_id=self.provider.name
+        ).exists()
+
+        # TODO(lb): check issues details page and see that integration shows in linked issues
+        return installation_element
+
+    def assert_has_installation_button(self, provider_row_element):
+        self.assert_button_element(
+            button=provider_row_element.install_button,
+            label='Install',
+            disabled=False,
+            icon='#icon-circle-add'
+        )
+
+    def assert_installation_element(self, installation_element):
+        self.assert_button_element(
+            button=installation_element.configure_button,
+            icon='#icon-settings',
+            label='Configure',
+            disabled=False,
+        )
+        self.assert_button_element(
+            button=installation_element.remove_button,
+            icon='#icon-trash',
+            label='Remove',
+            disabled=False,
+        )
+
+    def assert_installation_details_modal(self, installation_modal):
+        self.assert_button_element(
+            button=installation_modal.cancel_button,
+            label='Cancel'
+        )
```

Comment:
Is the text of the button an important part of the flow? I think ensuring that the dialog window closes would be something we could only do in an acceptance test.